Cython/Coverage.py

```diff
@@ -41,6 +41,15 @@ def _find_dep_file_path(main_file, file_path, relative_path_search=False):
         rel_file_path = os.path.join(os.path.dirname(main_file), file_path)
         if os.path.exists(rel_file_path):
             abs_path = os.path.abspath(rel_file_path)
+
+        # if file_path matches the main_file ending, that's it:
+        matching_abs_path = ''.join([os.path.splitext(main_file)[0], os.path.splitext(file_path)[1]])
+        if (
+                matching_abs_path.endswith(file_path)
```

Comment:
This may lead to false positives if file/directory names prefix others. It should take directory separators into account.

Reply:
Fair enough.

Reply:
@scoder Pathlib has some nice functions that could create a working fix for this but it is not included in python2.
I am willing to create a fix using the proper tools, but I need to know what cython's stance is on python2 support, otherwise it does not make sense.

Reply:
Cython 3.0 will support Python 2.7/3.4+, Cython 3.1 will drop support for Python 2.7 and 3.4. Potentially also for 3.5, but that isn't decided yet.

Reply:
Thanks! Okay so it makes sense to work on this patch, and it will probably be included from Cython 3.1 onwards as it won't build with python 2.7. That is acceptable.

My 2cents on the 3.5 support is to drop any versions which are not officially supported anymore. Also there have been quite some syntax and usability changes from 3.5 -> 3.6. 3.6 for instance supports the `os.PathLike` abstract class which makes it much easier when using Pathlib a lot. Also type annotation is much easier as you can do inline variable annotation. Also `f"{my_variable}"` instead of `"{0}".format(my_variable)`, and modern class declarations. Such as:
```Python
class example:
    my_var: int
```
It is quite a useful syntax toolbox that you get with 3.6. It makes python more pythonic. It is quite annoying to make do without these tools just to support a *deprecated* version of python.

Reply:
> It is quite annoying to make do without these tools just to support a _deprecated_ version of python.

Welcome to the world of a maintainer.